src/coreclr/jit/loopcloning.cpp

```diff
@@ -2969,11 +2969,18 @@ PhaseStatus Compiler::optCloneLoops()
 
     if (optLoopsCloned > 0)
     {
-        fgRenumberBlocks();
-
         fgInvalidateDfsTree();
         m_dfsTree = fgComputeDfs();
         m_loops   = FlowGraphNaturalLoops::Find(m_dfsTree);
+
+        if (optCanonicalizeLoops())
+        {
+            fgInvalidateDfsTree();
+            m_dfsTree = fgComputeDfs();
+            m_loops   = FlowGraphNaturalLoops::Find(m_dfsTree);
+        }
```

Comment:
Maybe one way to improve TP slightly would be to canonicalize exits as part of loop cloning. I might look into what this would save, but I would expect not that much since we're only paying here in cases where we actually did do loop cloning.

Reply:
These costs are likely dwarfed by the cost of cloning, so if we improve the cloning heuristics (which I assume will mean cloning fewer big loops) we should come out ahead.

Reply:
Seems like it wouldn't be hard for cloning to also clone all its exit blocks?

Reply:
It would work to clone the exits, although at the tradeoff of duplicating even more IR. Not sure whether that's worth it.